[Cases] Refactor notification subscriptions to case-subscribers logic, in API

dkarkanas opened the pull request:
- Add missing endpoint for case subscriptions

TODO:
- [x] api in the format of `/api/cases/{caseId}/subscribers`

dkarkanas committed:
[Cases] Add GetNotificationSubscriptions to ICasesManager wf integration

NikosDevPhp commented on src/Indice.Features.Cases.Workflows/Integrations/CasesManagerHttpClient.cs:
```diff
@@ -615,81 +508,86 @@ public virtual System.Threading.Tasks.Task AttachFileAsync(System.Guid caseId, F
-                    if (comment != null)
-                    {
+                    if (comment == null)
+                        throw new System.ArgumentNullException("comment");
```
Do we consider everything mandatory here?

dkarkanas commented:
todo api:

`cases/{caseId}/subscriptions`

dkarkanas committed:
Merge branch 'develop' of https://github.com/indice-co/Indice.Platform into feature/case-management/more-integration

dkarkanas committed:
Merge branch 'develop' of https://github.com/indice-co/Indice.Platform into feature/case-management/more-integration

# Conflicts:
#	src/Indice.Features.Cases.Workflows/Integrations/CasesManagerHttp.cs

dkarkanas committed:
[cases] update http client for workflows

dkarkanas committed:
[Cases] Refactor notification subscriptions

Refactored notification subscription methods to support pagination, replacing `GetSubscriptions` with `GetSubscribers` across interfaces and implementations. Updated API endpoints to focus on case-specific subscribers, introducing `{caseId}/subscribers`. Added `NotificationSubscriptionResultSet` for paginated results.

dkarkanas committed:
Merge branch 'develop' into feature/case-management/more-integration

dkarkanas committed:
[Cases] Add GetCaseTypeSubscribers integration api

dkarkanas committed:
Update API to .NET 8.0 and enhance OpenAPI spec

- Upgraded runtime to .NET 8.0 in `cases-management-api.nswag`.
- Updated OpenAPI spec with new URL, schemas, and metadata.
- Added `Subscriber` model for encapsulating email and group data.
- Refactored endpoints to standardize responses (e.g., `204 No Content`).
- Introduced `CasesAttachmentLink` schema for file attachment responses.
- Improved API documentation and OAuth2 authentication setup.
- Removed legacy models and streamlined data structures.
- Regenerated NSwag client code to align with updated API structure.

cleftheris reviewed (approved):
🎉🚀

cleftheris committed:
Merge branch 'develop' into feature/case-management/more-integration
